Datasnap/Classes/Utilities/Datasnap.m

```diff
@@ -77,45 +83,45 @@ - (id)initWithApiKey:(NSString*)apiKey
         apiKeySecret:(NSString*)apiKeySecret
       organizationId:(NSString*)organizationId
            projectId:(NSString*)projectId
-      eventQueueSize:(NSInteger)eventNum
+           idfaOptIn:(bool)optIn
+               email:(NSString*)email
  andVendorProperties:(VendorProperties*)vendorProperties
 {
     self.vendorProperties = vendorProperties;
     if (self = [self init]) {
         self.organizationId = organizationId;
         self.projectId = projectId;
-        self.eventQueue = [[EventQueue alloc] initWithSize:eventNum];
         self.api = [[DatasnapAPI alloc] initWithKey:apiKey secret:apiKeySecret];
-        self.baseClient = [[BaseClient alloc] init];
+        self.baseClient = [BaseClient new];
         self.baseClient.projectId = projectId;
         self.baseClient.organizationId = organizationId;
+        self.optIn = optIn;
+        self.email = email;
     }
     [self initializeData];
     return self;
 }
-
 + (id)sharedClient
 {
     static Datasnap* sharedClient = nil;
```

Comment:
This seems weird to me... what happens after sharedClient is called more than once?  Will it be nil?